src/System.IO.Pipelines/MemoryPoolBlock.cs

```diff
@@ -39,7 +39,14 @@ protected MemoryPoolBlock(MemoryPool pool, MemoryPoolSlab slab, int offset, int
 
         public override int Length => _length;
 
-        public override Span<byte> Span => new Span<byte>(Slab.Array, _offset, _length);
+        public override Span<byte> Span
+        {
+            get
+            {
+                if (IsDisposed) ThrowHelper.ThrowObjectDisposedException(nameof(MemoryPoolBlock));
+                return new Span<byte>(Slab.Array, _offset, _length);
```

Comment:
Is there a perfromance difference between the Span ctor that takes array vs the one that takes a pointer? Might be worth filing and issue to find it out, i.e. don't block this PR on that investigation.